launchpad/deposit.gno

```diff
@@ -0,0 +1,578 @@
+		state.DepositsByUserProject[deposit.depositor][deposit.projectId] = []string{}
+	}
+	state.DepositsByUserProject[deposit.depositor][deposit.projectId] = append(
+		state.DepositsByUserProject[deposit.depositor][deposit.projectId], 
+		deposit.id,
+	)
+}
+
+// processFirstDeposit handles the first deposit for a project tier
+func processFirstDeposit(info ProjectTierInfo) (Tier, error) {
+	if info.Tier.totalParticipant != 0 {
+		return info.Tier, nil
+	}
+
+	tier := info.Tier
+	tier.started.height = info.Height
+	tier.started.time = info.CurrentTime
+	tier.tierAmountPerBlockX96 = calcProjectTiersRewardPerBlockX96(tier)
+
+	return tier, nil
+}
+
+// DepositGns deposit gns to the project's tier
+// gns will be locked in `launchpad` contract
+// xgns will be minted to the `launchpad` contract
+// Returns depositId
+// ref: https://docs.gnoswap.io/contracts/launchpad/launchpad_deposit.gno#depositgns
+func DepositGns(targetProjectTierId string, amount uint64) string {
+	common.IsHalted()
+
+	projectId, tierStr := getProjectIdAndTierFromTierId(targetProjectTierId)
+	project, tier, err := validateProjectTier(projectId, tierStr)
+	if err != nil {
+		panic(err)
+	}
+
+	checkDepositConditions(project)
+
+	en.MintAndDistributeGns()
+
+	calculateDepositReward()
+	project = projects[projectId] // get updates project
+	tier = getTier(project, tierStr) // get updates tier
+
+	// Update gov_staker to calculate project's recipient's reward
+	gs.SetAmountByProjectWallet(project.recipient, amount, true)
+
+	// gns will be locked in `launchpad` contract
+	gns.TransferFrom(
+		a2u(std.PrevRealm().Addr()),
+		a2u(std.Address(consts.LAUNCHPAD_ADDR)),
+		amount,
+	)
+	xgns.Mint(
+		a2u(std.Address(consts.LAUNCHPAD_ADDR)),
+		amount,
+	)
+
+	// Create deposit
+	info := ProjectTierInfo{
+		Project:     project,
+		Tier:        tier,
+		TierStr:     tierStr,
+		Height:      uint64(std.GetHeight()),
+		CurrentTime: uint64(time.Now().Unix()),
+	}
+
+	deposit, err := createDeposit(info, amount)
+	if err != nil {
+		panic(err)
+	}
+
+	// Update indices
+	deposits[deposit.id] = deposit
+	updateDepositIndices(deposit, &DepositState{
+		Deposits:              deposits,
+		DepositsByProject:     depositsByProject,
+		DepositsByUser:        depositsByUser,
+		DepositsByUserProject: depositsByUserByProject,
+	})
+
+	// Emit deposit event
+	prevAddr, prevPkgPath := getPrev()
+	std.Emit(
+		"DepositGns",
+		"prevAddr", prevAddr,
+		"prevRealm", prevPkgPath,
+		"targetProjectTierId", targetProjectTierId,
+		"amount", ufmt.Sprintf("%d", amount),
+		"depositId", deposit.id,
+		"claimableHeight", ufmt.Sprintf("%d", deposit.claimableHeight),
+		"claimableTime", ufmt.Sprintf("%d", deposit.claimableTime),
+	)
+
+	// Process first deposit if applicable
+	updatedTier, err := processFirstDeposit(info)
+	if err != nil {
+		panic(err)
+	}
+
+	if updatedTier.started.height == info.Height {
+		std.Emit(
+			"FirstDepoistForProjectTier", // ????
```

Comment:
what is ????

Reply:
It seems that a memo about needing to organize and rename event names later was committed accidentally together with the code.

https://github.com/gnoswap-labs/gnoswap/pull/451/commits/eb596b597cbe9e8f65f9b07c5802e3767773ddb1